File: notification/src/main/java/sample2/module/chiclaim/com/notification/MainActivity.java
package sample2.module.chiclaim.com.notification;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.view.View;
import android.widget.Toast;

/**
 * android:launchMode="singleInstance"和singleTask
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        String from = getIntent().getStringExtra("from");
        if (!TextUtils.isEmpty(from)) {
            Toast.makeText(this, from, Toast.LENGTH_SHORT).show();
        }
    }

    public void goSecondActivity(View view) {
        startActivity(new Intent(this, SecondActivity.class));
    }
}

File: kotlin_coroutine/src/main/java/com/chiclaim/coroutine/MainActivity.kt
package com.chiclaim.coroutine

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun coroutineBasic(view: View) {
        startActivity(Intent(this, BasicCoroutineActivity::class.java))
    }
}
